### PR TITLE
feat: add tests for config edge cases

### PR DESCRIPTION
**Pull request by toddbaert**
This PR adds tests for a few flagd configuration edge-cases:

- targeting rule returns `null` should fall through to `defaultVariant` and return reason `DEFAULT`
- invalid/erroneous JSON logic (should error, and cause use of application (SDK) default)
- targeting returns a variant which is not in variant definition (should error, and cause use of application (SDK)  default)
- invalid sem_ver, fractional, and string-cmp inputs (should return null/false)
- non-string variants returned from the targeting rule
- timestamps

Fixes: https://github.com/open-feature/flagd-testbed/issues/84
Fixes: https://github.com/open-feature/flagd-testbed/issues/74
Relates to: https://github.com/open-feature/flagd/pull/1041

### CODE REVIEW

File: flags/custom-ops.json [hunk 1]
```diff
@@ -38,7 +38,14 @@
...
               },
-              "postfix", "none"
+              "postfix", {
+                "if": [
+                  {
+                    "ends_with": [{"var": "id"}, 3]
+                  },
+                  "fail", "none"
+                ]
+              }
```
Comment (toddbaert) [inline flags/custom-ops.json:48]: Currently this works `3 ends_with 3` is falsy when they are both ints. We can decide if that's actually what we want.

File: gherkin/flagd-json-evaluator.feature [hunk 2]
```diff
@@ -61,3 +63,12 @@ Feature: flagd json evaluation
...
       | "4.0.0" | "none"  |
+
+  Scenario Outline: Errors and edge cases
+    When an integer flag with key <key> is evaluated with default value 13
+    Then the returned value should be <value>
+    Examples:
+      | key                                 | value |
+      | "error-targeting-flag"              | 2     |
+      | "missing-variant-targeting-flag"    | 2     |
+      | "non-string-variant-targeting-flag" | 2     |
```
Comment (toddbaert) [inline gherkin/flagd-json-evaluator.feature:74]: These all fail in the Java in-process provider, and pass in the Java rpc provider.

File: flags/custom-ops.json (hunk 1 above)
Comment (toddbaert) [reply flags/custom-ops.json:48]: I could add that to my pending changes [here](https://github.com/open-feature/flagd/pull/1041).

File: gherkin/flagd-json-evaluator.feature (hunk 2 above)
Comment (Kavindu-Dodan) [reply gherkin/flagd-json-evaluator.feature:74]: yeah, the in-process providers fallback to default value if targeting rule evaluation fail.

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]
Looks good

**Review by beeme1mr — APPROVED** [review]
